lib/b2c/sessions.ts

```diff
@@ -1,71 +1,465 @@
+  apple_oauth_factor?: AppleOAuthFactor;
+  webauthn_factor?: WebAuthnFactor;
+  authenticator_app_factor?: AuthenticatorAppFactor;
+  github_oauth_factor?: GithubOAuthFactor;
+  recovery_code_factor?: RecoveryCodeFactor;
+  facebook_oauth_factor?: FacebookOAuthFactor;
+  crypto_wallet_factor?: CryptoWalletFactor;
+  amazon_oauth_factor?: AmazonOAuthFactor;
+  bitbucket_oauth_factor?: BitbucketOAuthFactor;
+  coinbase_oauth_factor?: CoinbaseOAuthFactor;
+  discord_oauth_factor?: DiscordOAuthFactor;
+  figma_oauth_factor?: FigmaOAuthFactor;
+  git_lab_oauth_factor?: GitLabOAuthFactor;
+  instagram_oauth_factor?: InstagramOAuthFactor;
+  linked_in_oauth_factor?: LinkedInOAuthFactor;
+  shopify_oauth_factor?: ShopifyOAuthFactor;
+  slack_oauth_factor?: SlackOAuthFactor;
+  snapchat_oauth_factor?: SnapchatOAuthFactor;
+  spotify_oauth_factor?: SpotifyOAuthFactor;
+  steam_oauth_factor?: SteamOAuthFactor;
+  tik_tok_oauth_factor?: TikTokOAuthFactor;
+  twitch_oauth_factor?: TwitchOAuthFactor;
+  twitter_oauth_factor?: TwitterOAuthFactor;
+  embeddable_magic_link_factor?: EmbeddableMagicLinkFactor;
+  biometric_factor?: BiometricFactor;
+  saml_sso_factor?: SAMLSSOFactor;
+  oidc_sso_factor?: OIDCSSOFactor;
+  salesforce_oauth_factor?: SalesforceOAuthFactor;
+  yahoo_oauth_factor?: YahooOAuthFactor;
+}
+
+export interface AuthenticatorAppFactor {
+  totp_id: string;
+}
+
+export interface BiometricFactor {
+  biometric_registration_id: string;
+}
+
+export interface BitbucketOAuthFactor {
+  id: string;
+  email_id: string;
+  provider_subject: string;
+}
+
+export interface CoinbaseOAuthFactor {
+  id: string;
+  email_id: string;
+  provider_subject: string;
+}
+
+export interface CryptoWalletFactor {
+  crypto_wallet_id: string;
+  crypto_wallet_address: string;
+  crypto_wallet_type: string;
+}
+
+export interface DiscordOAuthFactor {
+  id: string;
+  email_id: string;
+  provider_subject: string;
+}
+
+export interface EmailFactor {
+  email_id: string;
+  email_address: string;
+}
+
+export interface EmbeddableMagicLinkFactor {
+  embedded_id: string;
+}
+
+export interface FacebookOAuthFactor {
+  id: string;
+  email_id: string;
+  provider_subject: string;
+}
+
+export interface FigmaOAuthFactor {
+  id: string;
+  email_id: string;
+  provider_subject: string;
+}
+
+export interface GitLabOAuthFactor {
+  id: string;
+  email_id: string;
+  provider_subject: string;
+}
+
+export interface GithubOAuthFactor {
+  id: string;
+  email_id: string;
+  provider_subject: string;
+}
+
+export interface GoogleOAuthFactor {
+  id: string;
+  email_id: string;
+  provider_subject: string;
+}
+
+export interface InstagramOAuthFactor {
+  id: string;
+  email_id: string;
+  provider_subject: string;
 }
 
 export interface JWK {
-  alg: string;
-  key_ops: string[];
-  kid: string;
   kty: string;
   use: string;
+  key_ops: string[];
+  alg: string;
+  kid: string;
   x5c: string[];
-  "x5t#S256": string;
-
+  x5tS256: string;
```

Comment:
huh - I wonder what happened with the `#` in the field name? I _vaguely_ remember something about GRPC gateway having trouble serializing it to JSON, but I don't want to think too hard...